### PR TITLE
feat: Allow methods defined in a contract to be non-entry points

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

Related to #2688 

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by Maddiaa0 — APPROVED** [review]
lgtm

File: compiler/noirc_frontend/src/hir/def_map/mod.rs [hunk 1]
```diff
@@ -155,13 +155,23 @@ impl CrateDefMap {
...
             .iter()
             .filter_map(|(id, module)| {
                 if module.is_contract {
-                    let functions =
+                    let function_ids: Vec<FuncId> =
                         module.value_definitions().filter_map(|id| id.as_function()).collect();
+
+                    let functions = function_ids
+                        .into_iter()
+                        .map(|id| {
```
Comment (jfecher) [inline compiler/noirc_frontend/src/hir/def_map/mod.rs:168]: We shouldn't `collect` into `function_ids` then call `into_iter` immediately afterward, the intermediate `Vec` is not needed.

**Comment by Savio-Sou** [comment]
cc @noir-lang/developerrelations @rahul-kothari not sure if this needs to be documented on aztec.nr's end.

**Comment by critesjosh** [comment]
I'm not sure, I'll defer to @rahul-kothari

**Pull request merged**